### PR TITLE
fulfillments referrals update

### PR DESCRIPTION
**Pull request by VladMogwai**

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -1,2 +1,6 @@
 export const DEFAULT_QUERY_LIMIT = 50;
 export const MAXIMUM_QUERY_LIMIT = 500;
+export const DOMAINS = {
```
Comment (magonzalez) [inline src/constants.js]: This should be in env.yml, not here.

File: src/services/FulfillmentService.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
 /* eslint-disable no-unused-vars */
 import AWS from 'aws-sdk';
+import fetch from 'node-fetch';
+import { DOMAINS } from '../constants';
 import logger from '../utils/logger';
 import staticResources from '../models/staticResources';
-import referrals from '../../collections/referral/data';
+// import referrals from '../../collections/referral/data';
```
Comment (magonzalez) [inline src/services/FulfillmentService.js]: Should remove this and the file it refers to.

File: src/services/FulfillmentService.js [hunk 3]
```diff
@@ -47,10 +49,18 @@ class FulfillmentService {
         params: { slug },
         body: { type, value },
       } = req;
-
+      const url = `${DOMAINS[node.env.NODE_ENV]}/rest/v1/fulfillments/referrals/${slug.zipCode}`;
       // TODO: get items by slug,type and value
```
Comment (magonzalez) [inline src/services/FulfillmentService.js]: remove this

File: src/services/FulfillmentService.js [hunk 4]
```diff
@@ -47,10 +49,18 @@ class FulfillmentService {
         params: { slug },
         body: { type, value },
       } = req;
-
+      const url = `${DOMAINS[node.env.NODE_ENV]}/rest/v1/fulfillments/referrals/${slug.zipCode}`;
```
Comment (magonzalez) [inline src/services/FulfillmentService.js]: This method is supposed to get referrals by slug, as the name of the function implies. The slug will tell us what "category" of referral the user is asking for. I'm not sure why Eugene called the other values type and slug, but they should just be a location. For now, we should assume that location is the name of the county that the end user is in. So, the request should be for a particular type of referral in a particular county. This is complicated by the fat that there are two different categories of referrals. we have VCSO and MVPN. Let's chat about this.

File: src/services/FulfillmentService.js [hunk 5]
```diff
@@ -45,12 +45,24 @@ class FulfillmentService {
...
       } = req;
 
-      // TODO: get items by slug,type and value
-
-      return referrals;
+      const requestParams = {
+        type: slug.split('/')[1] || '',
+        county: location || '',
+      };
+      const searchValue = Object.values(requestParams).join('/');
```
Comment (magonzalez) [inline src/services/FulfillmentService.js]: This seems cryptic to me, relying on the order of fields defined in the requestParams object. I would rather see it be a little more straightforward.

```
const type = slug.split('/')[1] || '';
const county = location || '';
const url = `${process.env.PROJECT_URL}/rest/v1/fulfillments/referrals/${type}/${county}`;
```

Comment (VladMogwai) [reply src/services/FulfillmentService.js]: is it possible that user will not provide some of this params ?
for example we will get only location or only type ?

**Pull request merged**